src/main/webapp/scripts/tests/controller-auth-test.js

```diff
@@ -0,0 +1,93 @@
+import * as view from '/scripts/view.js';
+import * as controller from '/scripts/controller-auth.js';
+
+// Test (controller-auth.js and) view.js
+QUnit.module('Authorization controller', {
+  beforeEach: () => {
+    gapi = new CustomGapi();
+  },
+  afterEach: () => {
+    view.updateSigninStatus(false, null, null, null);
+    document.getElementById('result-text').innerText = '';
+    sinon.restore();
+  }
+});
+
+QUnit.test('When init fails, displays the authorization button and the error message', async function (assert) {
+  const errorMsg = 'error message';
+  sinon.stub(gapi.client, 'init').returns(Promise.reject(new Error(errorMsg)));
+
+  await controller.initClient();
+
+  assert.dom('#result-text').hasProperty('innerText', 'Failed to initialize client');
+});
+
+QUnit.test('When signed in, there should be sign-out and comment buttons', function (assert) {
+  const fakeSignOut = sinon.fake();
+  const fakeHandleCommentClick = sinon.fake();
+
+  view.updateSigninStatus(true, () => { }, fakeSignOut, fakeHandleCommentClick);
```

Comment:
Is it possible to test this by mocking the gapi client init to immediately return a resolved promised and gapi.auth2.getAuthInstance().isSignedIn.get() to return true so it could call updateSigninStatus through its natural flow?

Reply:
Similar comment for every test that directly uses the view.

Reply:
Done.